apps/src/templates/teacherDashboard/SectionActionBox.jsx

```diff
@@ -0,0 +1,216 @@
+      });
+  };
+
+  onClickEdit = () => {
+      this.props.handleEdit(this.props.sectionData.id);
+  };
+
+  onClickHideShow = () => {
+      this.props.toggleSectionHidden(this.props.sectionData.id);
+  };
+
+  onClickSync = () => {
+    // Section code is the course ID, without the G- or C- prefix.
+    const courseId = this.props.sectionCode.replace(/^[GC]-/, '');
+    this.props.updateRoster(courseId, this.props.sectionName)
+      .then(() => {
+        // While we are embedded in an angular page, reloading is the easiest
+        // way to pick up roster changes.  Once everything is React maybe we
+        // won't need to do this.
+        utils.reload();
+      });
+  };
+
+  onRequestDelete = () => {
+    this.setState({deleting: true});
+  };
+
+  render() {
+    const {sectionData} = this.props;
+    return (
+      <div style={{...styles.actionBox, ...this.props.style}}>
+        <a href={pegasus('/teacher-dashboard#/sections/' + sectionData.id)}>
+          <div style={styles.actionText}>
+            {i18n.sectionViewProgress()}
+          </div>
+        </a>
+        <a href={pegasus('/teacher-dashboard#/sections/' + sectionData.id + "/manage")}>
+          <div style={styles.actionText}>
+            {i18n.manageStudents()}
+          </div>
+        </a>
+        <a href={pegasus('/teacher-dashboard#/sections/' + sectionData.id + '/print_signin_cards')}>
+          <div style={styles.actionText}>
+            {i18n.printLoginCards()}
+          </div>
+        </a>
+        <a>
+          <div style={styles.actionTextBreak} onClick={this.onClickEdit} id="ui-test-edit-section">
+            {i18n.editSectionDetails()}
+          </div>
+        </a>
+        <a>
+          <div style={styles.actionText}>
+            <PrintCertificates
+              sectionId={sectionData.id}
+              assignmentName={sectionData.assignmentNames[0]}
+            />
+          </div>
+        </a>
+        {sectionData.loginType === OAuthSectionTypes.clever &&
+          <a>
+            <div style={styles.actionText} onClick={this.onClickSync}>
+              {i18n.syncClever()}
+            </div>
+          </a>
+        }
+        {sectionData.loginType === OAuthSectionTypes.google_classroom &&
+          <a>
+            <div style={styles.actionText} onClick={this.onClickSync}>
+              {i18n.syncGoogleClassroom()}
+            </div>
+          </a>
+        }
+        <a>
+          <div style={styles.actionText} onClick={this.onClickHideShow} id="ui-test-hide-section">
+            {sectionData.hidden ? i18n.showSection() : i18n.hideSection()}
+          </div>
+        </a>
+        {sectionData.studentCount === 0 && (
+          <div>
+            <a>
+              <div style={styles.archiveDelete} onClick={this.onRequestDelete}>
+                <FontAwesome icon=" fa-times-circle" style={styles.xIcon}/>
+                {i18n.deleteSection()}
+              </div>
+            </a>
+            <BaseDialog
```

Comment:
Even if you leave it at the end of this file, it'd help readability to extract this delete confirmation dialog as its own component.  Fortunately there's very little in the way of props you'd need to pass through.